### PR TITLE
powerdns-admin: init at 0.2.3

### PR DESCRIPTION
**Pull request by zhaofengli**
###### Motivation for this change

[PowerDNS-Admin](https://github.com/ngoduykhanh/PowerDNS-Admin) is a web interface for PowerDNS. This PR also adds new Python modules required by the application.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by zhaofengli** [comment]
Rebased with `stdenv.lib` fixes for #108938. Also fixed font copying in asset packaging.

File: pkgs/applications/networking/powerdns-admin/update-asset-deps.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -I nixpkgs=../../../.. -i bash -p wget yarn2nix-moretea.yarn2nix jq
+
+# This script is based upon:
+# pkgs/applications/networking/instant-messengers/riot/update-riot-desktop.sh
+
+set -euo pipefail
+
+if [ "$#" -ne 1 ] || [[ "$1" == -* ]]; then
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/powerdns-admin/update-asset-deps.sh:9]:
```suggestion
if [[ $# -ne 1 || $1 == -* ]]; then
```

File: pkgs/development/python-modules/xmlsec/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    sha256 = "1c4k42zv3plm6v65p7z6l5rxyf50h40d02nhc16vq7cjrfvdf4rx";
+  };
+
+  # https://github.com/mehcode/python-xmlsec/issues/84#issuecomment-632930116
+  patches = [
+    ./reset-lxml-in-tests.patch
+  ];
+
+  buildInputs = [
+    pkgs.xmlsec pkgs.libxslt pkgs.libxml2 pkgs.libtool
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/xmlsec/default.nix:19]: Please inherit them from python.nix.

File: pkgs/development/python-modules/xmlsec/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  ];
+  nativeBuildInputs = [
+    pkgs.pkg-config pkgconfig setuptools_scm toml
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/xmlsec/default.nix:22]:
```suggestion
    pkg-config pkgconfig setuptools-scm toml
```

File: pkgs/development/python-modules/xmlsec/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  ];
+  propagatedBuildInputs = [ lxml ];
+
+  # Full git clone required for test_doc_examples
+  checkInputs = [ pytestCheckHook hypothesis ];
+  pytestFlagsArray = [ "tests/" "--ignore=tests/test_doc_examples.py" ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/xmlsec/default.nix:28]:
```suggestion
  pytestFlagsArray = [ "--ignore=tests/test_doc_examples.py" ];
```

set automatically

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/xmlsec/default.nix (hunk 2 above)
Comment (zhaofengli) [reply pkgs/development/python-modules/xmlsec/default.nix:19]: Sorry, could you elaborate on what you mean by that?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109841` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
16 packages built:
</summary>
<ul>
<li>powerdns-admin</li>
<li>python37Packages.flask-seasurf</li>
<li>python37Packages.flask-sslify</li>
<li>python37Packages.lima</li>
<li>python37Packages.python3-saml</li>
<li>python37Packages.xmlsec</li>
<li>python38Packages.flask-seasurf</li>
<li>python38Packages.flask-sslify</li>
<li>python38Packages.lima</li>
<li>python38Packages.python3-saml</li>
<li>python38Packages.xmlsec</li>
<li>python39Packages.flask-seasurf</li>
<li>python39Packages.flask-sslify</li>
<li>python39Packages.lima</li>
<li>python39Packages.python3-saml</li>
<li>python39Packages.xmlsec</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
powerdns-admin:
<details>


`patchPhase` should not be overridden, use `postPatch` instead.

Near pkgs/applications/networking/powerdns-admin/default.nix:80:3:

```
   |
80 |   patchPhase = ''
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/patch-phase.md
`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/networking/powerdns-admin/default.nix:84:3:

```
   |
84 |   installPhase = ''
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
python37Packages.flask-seasurf:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
Unused argument: `stdenv`.
Near pkgs/development/python-modules/flask-seasurf/default.nix:1:8:

```
  |
1 | { lib, stdenv, fetchFromGitHub, buildPythonPackage, flask }:
  |        ^
```
</details>
python37Packages.flask-sslify:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
Unused argument: `stdenv`.
Near pkgs/development/python-modules/flask-sslify/default.nix:1:8:

```
  |
1 | { lib, stdenv, fetchPypi, buildPythonPackage, flask }:
  |        ^
```
</details>
python37Packages.python3-saml:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
Unused argument: `stdenv`.
Near pkgs/development/python-modules/python3-saml/default.nix:1:8:

```
  |
1 | { lib, stdenv, fetchFromGitHub, buildPythonPackage, isPy3k,
  |        ^
```
</details>
python37Packages.xmlsec:
<details>


Unused argument: `stdenv`.
Near pkgs/development/python-modules/xmlsec/default.nix:1:14:

```
  |
1 | { pkgs, lib, stdenv, fetchPypi, buildPythonPackage, pytestCheckHook,
  |              ^
```
</details>
python38Packages.flask-seasurf:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
Unused argument: `stdenv`.
Near pkgs/development/python-modules/flask-seasurf/default.nix:1:8:

```
  |
1 | { lib, stdenv, fetchFromGitHub, buildPythonPackage, flask }:
  |        ^
```
</details>
python38Packages.flask-sslify:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
Unused argument: `stdenv`.
Near pkgs/development/python-modules/flask-sslify/default.nix:1:8:

```
  |
1 | { lib, stdenv, fetchPypi, buildPythonPackage, flask }:
  |        ^
```
</details>
python38Packages.python3-saml:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
Unused argument: `stdenv`.
Near pkgs/development/python-modules/python3-saml/default.nix:1:8:

```
  |
1 | { lib, stdenv, fetchFromGitHub, buildPythonPackage, isPy3k,
  |        ^
```
</details>
python38Packages.xmlsec:
<details>


Unused argument: `stdenv`.
Near pkgs/development/python-modules/xmlsec/default.nix:1:14:

```
  |
1 | { pkgs, lib, stdenv, fetchPypi, buildPythonPackage, pytestCheckHook,
  |              ^
```
</details>
python39Packages.flask-seasurf:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
Unused argument: `stdenv`.
Near pkgs/development/python-modules/flask-seasurf/default.nix:1:8:

```
  |
1 | { lib, stdenv, fetchFromGitHub, buildPythonPackage, flask }:
  |        ^
```
</details>
python39Packages.flask-sslify:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
Unused argument: `stdenv`.
Near pkgs/development/python-modules/flask-sslify/default.nix:1:8:

```
  |
1 | { lib, stdenv, fetchPypi, buildPythonPackage, flask }:
  |        ^
```
</details>
python39Packages.python3-saml:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
Unused argument: `stdenv`.
Near pkgs/development/python-modules/python3-saml/default.nix:1:8:

```
  |
1 | { lib, stdenv, fetchFromGitHub, buildPythonPackage, isPy3k,
  |        ^
```
</details>
python39Packages.xmlsec:
<details>


Unused argument: `stdenv`.
Near pkgs/development/python-modules/xmlsec/default.nix:1:14:

```
  |
1 | { pkgs, lib, stdenv, fetchPypi, buildPythonPackage, pytestCheckHook,
  |              ^
```
</details>
</details>

**Comment by zhaofengli** [comment]
Ok, force-pushed with the requested changes. For the xmlsec, I agree it's a little messy, but I'm not entirely sure how to get it to compile without having the native packages pulled in (both `pkgs.pkg-config` as well as `pythonPackages.pkgconfig` are required in my testing).

**Comment by SuperSandro2000** [comment]
> Ok, force-pushed with the requested changes. For xmlsec, I agree it's a little messy, but I'm not entirely sure how to get it to compile without having the native packages pulled in (both `pkgs.pkg-config` as well as `pythonPackages.pkgconfig` are required in my testing).

Probably the package is using pkg-config somewhere directly instead of through pkgconfig. Should not be a blocker.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109841` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
15 packages built:
</summary>
<ul>
<li>python37Packages.flask-seasurf</li>
<li>python37Packages.flask-sslify</li>
<li>python37Packages.lima</li>
<li>python37Packages.python3-saml</li>
<li>python37Packages.xmlsec</li>
<li>python38Packages.flask-seasurf</li>
<li>python38Packages.flask-sslify</li>
<li>python38Packages.lima</li>
<li>python38Packages.python3-saml</li>
<li>python38Packages.xmlsec</li>
<li>python39Packages.flask-seasurf</li>
<li>python39Packages.flask-sslify</li>
<li>python39Packages.lima</li>
<li>python39Packages.python3-saml</li>
<li>python39Packages.xmlsec</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
python37Packages.flask-seasurf:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python37Packages.flask-sslify:
<details>


Unused argument: `pytestCheckHook`.
Near pkgs/development/python-modules/flask-sslify/default.nix:1:39:

```
  |
1 | { lib, fetchPypi, buildPythonPackage, pytestCheckHook, flask }:
  |                                       ^
```
</details>
python37Packages.python3-saml:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python38Packages.flask-seasurf:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python38Packages.flask-sslify:
<details>


Unused argument: `pytestCheckHook`.
Near pkgs/development/python-modules/flask-sslify/default.nix:1:39:

```
  |
1 | { lib, fetchPypi, buildPythonPackage, pytestCheckHook, flask }:
  |                                       ^
```
</details>
python38Packages.python3-saml:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python39Packages.flask-seasurf:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python39Packages.flask-sslify:
<details>


Unused argument: `pytestCheckHook`.
Near pkgs/development/python-modules/flask-sslify/default.nix:1:39:

```
  |
1 | { lib, fetchPypi, buildPythonPackage, pytestCheckHook, flask }:
  |                                       ^
```
</details>
python39Packages.python3-saml:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
</details>

**Comment by zhaofengli** [comment]
Thanks! I removed the unused parameter from flask-sslify. The other "Add a checkPhase for tests" errors seem to be false-positives because tests are indeed run when they are built.

File: pkgs/development/python-modules/xmlsec/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/xmlsec/default.nix:19]: in ``pkgs/top-level/python-packages.nix``:
```nix
wxPython30 = callPackage ../development/python-modules/wxPython/3.0.nix {
  inherit (pkgs) libtool libxml2 pkg-config xmlsec;
};
```

File: pkgs/development/python-modules/python3-saml/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    sha256 = "1wh8rhjmjsl7y6fknzhrwwr69say68m11fqmywvxijk9kpn6qavl";
+  };
+
+  propagatedBuildInputs = [
+    isodate lxml xmlsec defusedxml
+  ];
+
+  checkInputs = [ freezegun ];
+
+  disabled = !isPy3k;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/python3-saml/default.nix:21]:
```suggestion
```

File: pkgs/development/python-modules/python3-saml/default.nix [hunk 6]
```diff
@@ -0,0 +1,29 @@
+{ lib, fetchFromGitHub, buildPythonPackage, isPy3k,
+isodate, lxml, xmlsec, defusedxml, freezegun }:
+
+buildPythonPackage rec {
+  pname = "python3-saml";
+  version = "1.10.0";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/python3-saml/default.nix:6]:
```suggestion
  version = "1.10.0";
  disabled = !isPy3k;
```

File: pkgs/applications/networking/powerdns-admin/update-asset-deps.sh [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+# This script is based upon:
+# pkgs/applications/networking/instant-messengers/riot/update-riot-desktop.sh
+
+set -euo pipefail
+
+if [[ $# -ne 1 || $1 == -* ]]; then
+        echo "Regenerates the Yarn dependency lock files for the powerdns-admin package."
+        echo "Usage: $0 <git release version>"
+        exit 1
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/powerdns-admin/update-asset-deps.sh:12]:
```suggestion
    echo "Regenerates the Yarn dependency lock files for the powerdns-admin package."
    echo "Usage: $0 <git release version>"
    exit 1
```

File: pkgs/development/python-modules/xmlsec/default.nix [hunk 8]
```diff
@@ -0,0 +1,36 @@
+{ pkgs, lib, fetchPypi, buildPythonPackage, pytestCheckHook,
+pkgconfig, setuptools_scm, toml, lxml, hypothesis }:
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/xmlsec/default.nix:2]:
```suggestion
pkgconfig, setuptools-scm, toml, lxml, hypothesis }:
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/xmlsec/default.nix (hunk 8 above)
Comment (zhaofengli) [reply pkgs/development/python-modules/xmlsec/default.nix:2]: The attribute is named `setuptools_scm`, and `setuptools-scm` doesn't exist. Or am I missing what you are trying to say here?

File: pkgs/development/python-modules/xmlsec/default.nix [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+  # https://github.com/mehcode/python-xmlsec/issues/84#issuecomment-632930116
+  patches = [
+    ./reset-lxml-in-tests.patch
+  ];
+
+  buildInputs = [
+    xmlsec libxslt libxml2 libtool
+  ];
+  nativeBuildInputs = [
+    pkg-config pkgconfig setuptools_scm toml
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/xmlsec/default.nix:35]:
```suggestion
    pkg-config pkgconfig setuptools-scm toml
```

File: pkgs/development/python-modules/xmlsec/default.nix [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+, fetchPypi
+, buildPythonPackage
+, pytestCheckHook
+, libxslt
+, libxml2
+, libtool
+, pkg-config
+, xmlsec
+, pkgconfig
+, setuptools_scm
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/xmlsec/default.nix:11]:
```suggestion
, setuptools-scm
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/xmlsec/default.nix (hunk 8 above)
Comment (zhaofengli) [reply pkgs/development/python-modules/xmlsec/default.nix]: Ah nevermind, I was looking at the wrong place.

**Comment by zhaofengli** [comment]
Well, I was looking at the wrong place :facepalm: Fix applied.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109841` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
15 packages built:
</summary>
<ul>
<li>python37Packages.flask-seasurf</li>
<li>python37Packages.flask-sslify</li>
<li>python37Packages.lima</li>
<li>python37Packages.python3-saml</li>
<li>python37Packages.xmlsec</li>
<li>python38Packages.flask-seasurf</li>
<li>python38Packages.flask-sslify</li>
<li>python38Packages.lima</li>
<li>python38Packages.python3-saml</li>
<li>python38Packages.xmlsec</li>
<li>python39Packages.flask-seasurf</li>
<li>python39Packages.flask-sslify</li>
<li>python39Packages.lima</li>
<li>python39Packages.python3-saml</li>
<li>python39Packages.xmlsec</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
python37Packages.flask-seasurf:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python37Packages.python3-saml:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python38Packages.flask-seasurf:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python38Packages.python3-saml:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python39Packages.flask-seasurf:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python39Packages.python3-saml:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
</details>

**Comment by zhaofengli** [comment]
Alright, added. I thought those were not necessary as tests are run already, but doesn't hurt to have them I guess :stuck_out_tongue:

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109841` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
16 packages built:
</summary>
<ul>
<li>powerdns-admin</li>
<li>python37Packages.flask-seasurf</li>
<li>python37Packages.flask-sslify</li>
<li>python37Packages.lima</li>
<li>python37Packages.python3-saml</li>
<li>python37Packages.xmlsec</li>
<li>python38Packages.flask-seasurf</li>
<li>python38Packages.flask-sslify</li>
<li>python38Packages.lima</li>
<li>python38Packages.python3-saml</li>
<li>python38Packages.xmlsec</li>
<li>python39Packages.flask-seasurf</li>
<li>python39Packages.flask-sslify</li>
<li>python39Packages.lima</li>
<li>python39Packages.python3-saml</li>
<li>python39Packages.xmlsec</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
python37Packages.flask-seasurf:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python37Packages.python3-saml:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python38Packages.flask-seasurf:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python38Packages.python3-saml:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python39Packages.flask-seasurf:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python39Packages.python3-saml:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
</details>

**Comment by zhaofengli** [comment]
> python38Packages.flask-seasurf:
> Add a checkPhase for tests, or at least pythonImportsCheck.

Hmm, the `pythonImportsCheck`s are definitely there. Not sure what's happening here.

File: pkgs/development/python-modules/xmlsec/default.nix (hunk 8 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/xmlsec/default.nix]: It was added a few weeks ago and new contributions should use the - variant.

**Comment by SuperSandro2000** [comment]
> > python38Packages.flask-seasurf:
> > Add a checkPhase for tests, or at least pythonImportsCheck.
> 
> Hmm, the `pythonImportsCheck`s are definitely there. Not sure what's happening here.

It probably used the changes before you added the checks. I want to add a link to the commit that was used to generate the report but that maybe takes a bit. Taking a look at the other changes later.

File: pkgs/development/python-modules/xmlsec/default.nix [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+    xmlsec libxslt libxml2 libtool
+  ];
+  nativeBuildInputs = [
+    pkg-config pkgconfig setuptools-scm toml
+  ];
+  propagatedBuildInputs = [ lxml ];
+
+  # Full git clone required for test_doc_examples
+  checkInputs = [ pytestCheckHook hypothesis ];
+  pytestFlagsArray = [ "--ignore=tests/test_doc_examples.py" ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/xmlsec/default.nix:41]:
```suggestion
  disabledTestPaths = [ "tests/test_doc_examples.py" ];
```

File: pkgs/development/python-modules/xmlsec/default.nix [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+    ./reset-lxml-in-tests.patch
+  ];
+
+  buildInputs = [
+    xmlsec libxslt libxml2 libtool
+  ];
+  nativeBuildInputs = [
+    pkg-config pkgconfig setuptools-scm toml
+  ];
+  propagatedBuildInputs = [ lxml ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/xmlsec/default.nix:37]:
```suggestion
  nativeBuildInputs = [ pkg-config pkgconfig setuptools-scm toml ];

  buildInputs = [ xmlsec libxslt libxml2 libtool ];

  propagatedBuildInputs = [ lxml ];
```

**Comment by SuperSandro2000** [comment]
@zhaofengli please resolve the merge conflict.

**Comment by zhaofengli** [comment]
Sure, rebased and applied suggested changes.

**Comment by SuperSandro2000** [comment]
```
error: build of '/nix/store/75q369al977y8q8h9b1cgj6a5frn6vxn-python3.8-python3-saml-1.10.0.drv' on 'ssh://clara.r' failed: error: --- Error --- nix-daemon
builder for '/nix/store/75q369al977y8q8h9b1cgj6a5frn6vxn-python3.8-python3-saml-1.10.0.drv' failed with exit code 1; last 10 log lines:
  adding 'python3_saml-1.10.0.dist-info/RECORD'
  removing build/bdist.linux-x86_64/wheel
  Finished executing setuptoolsBuildPhase
  installing
  Executing pipInstallPhase
  /build/source/dist /build/source
  Processing ./python3_saml-1.10.0-py3-none-any.whl
  Requirement already satisfied: xmlsec>=1.0.5 in /nix/store/gknwvwa2g06r9l2469km6yb6qva8ffvj-python3.8-xmlsec-1.3.9/lib/python3.8/site-packages (from python3-saml==1.10.0) (1.3.9)
  ERROR: Could not find a version that satisfies the requirement defusedxml==0.6.0 (from python3-saml)
  ERROR: No matching distribution found for defusedxml==0.6.0
builder for '/nix/store/75q369al977y8q8h9b1cgj6a5frn6vxn-python3.8-python3-saml-1.10.0.drv' failed with exit code 1
cannot build derivation '/nix/store/61dykksav9ih12pa0bms62a7rlyj4s9z-powerdns-admin-assets-0.2.3.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/g9rpjamn7hmlcrq9fimmz2d03pvb50vf-powerdns-admin-0.2.3.drv': 2 dependencies couldn't be built
error: build of '/nix/store/g9rpjamn7hmlcrq9fimmz2d03pvb50vf-powerdns-admin-0.2.3.drv' failed
```

I think you need to relax the version requirement.

**Comment by zhaofengli** [comment]
@SuperSandro2000 Ok, looks like the upstream has [removed the dependency](https://github.com/onelogin/python3-saml/pull/236) and the commit doesn't introduce any API/functionality changes, so I've added the patch. Also took the liberty to bump it to 1.10.1.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109841` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
11 packages built:
</summary>
<ul>
<li>powerdns-admin</li>
<li>python38Packages.flask-seasurf</li>
<li>python38Packages.flask-sslify</li>
<li>python38Packages.lima</li>
<li>python38Packages.python3-saml</li>
<li>python38Packages.xmlsec</li>
<li>python39Packages.flask-seasurf</li>
<li>python39Packages.flask-sslify</li>
<li>python39Packages.lima</li>
<li>python39Packages.python3-saml</li>
<li>python39Packages.xmlsec</li>
</ul>
</details>

**Pull request merged**